lib/src/scip_visitor.dart

```diff
@@ -115,9 +115,12 @@ class ScipVisitor extends GeneralizingAstVisitor {
     // is a `CompoundAssignmentExpression`, we know this node is referring
     // to an assignment line. In that case, use the read/write element attached
     // to this node instead of the [node]'s element
-    if (node.parent is CompoundAssignmentExpression) {
-      final assignmentNode = node.parent as CompoundAssignmentExpression;
-      element = assignmentNode.readElement ?? assignmentNode.writeElement;
+    if (element == null) {
+      final assignmentExpr =
+          node.thisOrAncestorOfType<CompoundAssignmentExpression>();
+      if (assignmentExpr == null) return;
+
+      element = assignmentExpr.readElement ?? assignmentExpr.writeElement;
     }
```

Comment:
Given the following dart code:
```dart
Foo()..value = 1;
```

This is what the generated ast results in

```
Foo()..value = 1; (ExpressionStatement)
        Foo()..value = 1 (CascadeExpression)
                Foo() (MethodInvocation)
                        Foo (SimpleIdentifier)
                        () (ArgumentList)
                ..value = 1 (AssignmentExpression) <----- (extends from CompoundAssignmentExpression)
                        ..value (PropertyAccess)
                                value (SimpleIdentifier)
                        1 (IntegerLiteral)
```

Given the above code, `value (SimpleIdentifier)` is the case that makes it to this conditional. If we refer to the ast, the parent of this element is `PropertyAccess`, not `CompoundAssignmentExpression`, which tells us this expression can occur at multiple depths, not just the parent

We can solve this by just searching for the closest node of the specific type, and treat that as the assignment expression we care about